custom/icds_reports/reports/service_delivery_report.py

```diff
@@ -0,0 +1,168 @@
+                ('Number of beneficiaries who did not attend pre-school education', 'pse_0_days'),
+                ('Number of beneficiaries enrolled for Anganwadi services', 'pse_eligible'),
+                ('Percentage of beneficiaries who did not attend pre-school education', 'pse_0_days', 'pse_eligible'),
+                ('Number of beneficiaries who attended pre-school education for 1-7 days', 'pse_1_7_days'),
+                ('Number of beneficiaries enrolled for Anganwadi services', 'pse_eligible'),
+                ('Percentage of beneficiaries who attended pre-school education for 1-7 days', 'pse_1_7_days', 'pse_eligible'),
+                ('Number of beneficiaries who attended pre-school education for 8-14 days', 'pse_8_14_days'),
+                ('Number of beneficiaries enrolled for Anganwadi services', 'pse_eligible'),
+                ('Percentage of beneficiaries who attended pre-school education for 8-14 days', 'pse_8_14_days', 'pse_eligible'),
+                ('Number of beneficiaries who attended pre-school education for 15-20 days', 'pse_15_20_days'),
+                ('Number of beneficiaries enrolled for Anganwadi services', 'pse_eligible'),
+                ('Percentage of beneficiaries who attended pre-school education for 15-20 days', 'pse_15_20_days', 'pse_eligible'),
+                ('Number of beneficiaries who attended pre-school education for at least 21 days','pse_21_days'),
+                ('Number of beneficiaries enrolled for Anganwadi services', 'pse_eligible'),
+                ('Percentage of beneficiaries who attended pre-school education for at least 21 days', 'pse_21_days', 'pse_eligible'),
+                ('Number of children who were weighed', 'gm_3_5'),
+                ('Total children enrolled for Anganwadi services', 'children_3_5'),
+                ('Percentage of children who were weighed', 'gm_3_5', 'children_3_5')
+            ]
+
+        return headers
+
+    def get_excel_data(self):
+        def evaulate_value(row, headers_with_columns):
+            row_data = []
+
+            for header in headers_with_columns:
+                if len(header) == 2:
+                    row_data.append(row[header[1]])
+                else:
+                    if row[header[2]]:
+                        percentage = row[header[1]]/row[header[2]] * 100
+                    else:
+                        percentage = 0
+                    row_data.append("%.2f" % percentage)
+
+            return row_data
+
+        filters = copy.deepcopy(self.config)
+
+        del filters['beneficiary_category']
+        del filters['domain']
+
+        order_by = self.location_columns
+
+        values = {header[1] for header in self.headers_and_calculation if len(header) == 2}
+        headers = [header[0] for header in self.headers_and_calculation]
+
+        data = ServiceDeliveryReportView.objects.filter(**filters).order_by(*order_by).values(*values)
```

Comment:
can you pass a set like this to `values`?

Reply:
yeah it works